workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/ReaderWriterStore.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    internal class ReaderWriterStore
+    {
+        private readonly Dictionary<MonoBehaviour, Dictionary<uint, IReaderWriterInternal[]>> behaviourToReadersWriters
+            = new Dictionary<MonoBehaviour, Dictionary<uint, IReaderWriterInternal[]>>();
+
+        private readonly Dictionary<uint, HashSet<IReaderWriterInternal>> componentIdToReadersWriters =
+            new Dictionary<uint, HashSet<IReaderWriterInternal>>();
+
+        public void AddReaderWritersForBehaviour(MonoBehaviour behaviour,
+            Dictionary<uint, IReaderWriterInternal[]> componentIdsToReaderWriterLists)
+        {
+            behaviourToReadersWriters[behaviour] = componentIdsToReaderWriterLists;
```

Comment:
Is this allowed to overwrite if something is already there? if not, use `behaviourToReadersWriters.Add`